src/components/app/App.jsx

```diff
@@ -48,34 +59,36 @@ const App = () => {
         <NoticesProvider>
           <ToastsProvider>
             <Toasts />
-            <Switch>
-              {/* always remove trailing slashes from any route */}
-              <Redirect from="/:url*(/+)" to={global.location.pathname.slice(0, -1)} />
-              {/* page routes for the app */}
-              <AuthenticatedPageRoute exact path="/" component={EnterpriseCustomerRedirect} />
-              <AuthenticatedPageRoute exact path="/r/:redirectPath+" component={EnterprisePageRedirect} />
-              <PageRoute exact path="/invite/:enterpriseCustomerInviteKey" component={EnterpriseInvitePage} />
-              <PageRoute
-                exact
+            {/* always remove trailing slashes from any route */}
+            <TruncatedLocation />
+            {/* page routes for the app */}
+            <Routes>
+              <Route path="/" element={<AuthenticatedPageRoute><EnterpriseCustomerRedirect /></AuthenticatedPageRoute>} />
+              <Route path="/r/*" element={<AuthenticatedPageRoute><EnterprisePageRedirect /></AuthenticatedPageRoute>} />
+              <Route path="/invite/:enterpriseCustomerInviteKey" element={<PageWrap><EnterpriseInvitePage /></PageWrap>} />
+              <Route
                 path="/:enterpriseSlug/executive-education-2u"
-                render={(routeProps) => (
-                  <AuthenticatedPage>
-                    <ExecutiveEducation2UPage {...routeProps} />
-                  </AuthenticatedPage>
+                element={(
+                  <PageWrap>
+                    <AuthenticatedPage>
+                      <ExecutiveEducation2UPage />
+                    </AuthenticatedPage>
+                  </PageWrap>
                 )}
               />
-              <PageRoute
-                exact
+              <Route
                 path="/:enterpriseSlug/executive-education-2u/enrollment-completed"
-                render={(routeProps) => (
-                  <AuthenticatedPage>
-                    <EnrollmentCompleted {...routeProps} />
-                  </AuthenticatedPage>
+                element={(
+                  <PageWrap>
+                    <AuthenticatedPage>
+                      <EnrollmentCompleted />
+                    </AuthenticatedPage>
+                  </PageWrap>
                 )}
               />
-              <Route path="/:enterpriseSlug" component={EnterpriseAppPageRoutes} />
-              <PageRoute path="*" component={NotFoundPage} />
-            </Switch>
+              <Route path="/:enterpriseSlug/*" element={<EnterpriseAppPageRoutes />} />
```

Comment:
Curious: why do we need to add the * at the end?

Reply:
> Curious: why do we need to add the * at the end?

React Router V6 handles nested routes a little bit differently. So in order to match the descendent routes we have to add this `*`. [[ref](https://stackoverflow.com/questions/73756457/react-router-v6-issues-with-routes-as-child-component?rq=3#answer-73756945)]